src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -215,6 +216,7 @@
   "Delete the given host from the db"
   [certname]
   {:pre [certname]}
+  (jdbc/delete! :catalog_inputs ["certname_id in (select id from certnames where certname=?)" certname])
   (jdbc/delete! :certname_packages ["certname_id in (select id from certnames where certname=?)" certname])
   (jdbc/delete! :certnames ["certname=?" certname]))
```

Comment:
May not matter yet, but if we add more certname_id related lines like this and assuming correct transactional protection, wonder if we could avoid some work by returning the id from the first delete and using it for others.  i.e. `delete from foo where ... returning certname_id`, or assuming it's feasible and we can get it right (this likely isn't quite), do it all at once:

```
with ids as (delete from whatever where ... returning certid
                 union delete from ...)
    delete from certnames where id in (select certid from ids);
```

Not entirely sure this would be wise, just wondered.